### DISCUSSION TITLE
Keycloak SAML Integration with IBM Cloud Pak for Data

### DISCUSSION THREAD

**Discussion by Wh1terabit** [keycloak/keycloak#22756] [Q&A]
Hello everyone,

I'm facing an issue while integrating Keycloak SAML mappings with IBM Cloud Pak for Data (CP4D) on Red Hat OpenShift. We've successfully implemented the authentication part, but we're struggling with the mappings between Keycloak and CP4D, specifically related to First Name, Last Name, and Groups.

Our current setup involves creating a client scope between Keycloak and CP4D and configuring the necessary mappings. However, after the SAML authentication process, only the email address and username from Keycloak are being transferred to CP4D. The mappings for First Name, Last Name, and Groups are not working as expected, and we've already reached out to IBM support for assistance.

Following a thorough review of our CP4D configurations, the IBM support team has indicated that the issue might be on the SAML configuration side within Keycloak.

I would greatly appreciate any insights or advice on how to troubleshoot and resolve this matter. If anyone has experience with configuring SAML mappings between Keycloak and CP4D, especially related to transferring First Name, Last Name, and Group information, your guidance would be invaluable.

Thank you in advance for your assistance and expertise.

The mappings that we created is something like the following:

**"token_attribute_mappings": {
            "sub": "email",
            "given_name": "firstName",
            "family_name": "lastName",
            "groups":"Groups",
            "email": "email"
        }**

example mapping into  keycloak 

![image](https://github.com/keycloak/keycloak/assets/127850860/ff36672c-a39e-402f-a720-2fc707b03a08)

**Comment by sujit-kumar-singh** [comment]
Hi, You can follow this for the user properties/attributes mapper

https://josh-tracy.github.io/Ansible_Tower_RedHatSSO/#step-4-configure-mappers
